File: app/src/main/java/com/example/androidautoclicker/CustomEvent.java
package com.example.androidautoclicker;

import android.accessibilityservice.GestureDescription;
import android.graphics.Path;
import android.graphics.Point;

public abstract class CustomEvent {
    protected long startTime = 10L;
    protected long duration = 10L;
    protected Path path = new Path();

    public GestureDescription.StrokeDescription onEvent() {
        path.reset();
        createPath();
        return new GestureDescription.StrokeDescription(path, startTime, duration);
    }

    protected abstract void createPath();

    public void setStartTime(long startTime) {
        this.startTime = startTime;
    }

    public void setDuration(long duration) {
        this.duration = duration;
    }
}

class Click extends CustomEvent {
    private final Point point;

    public Click(Point point) {
        this.point = point;
    }

    @Override
    protected void createPath() {
        path.moveTo(point.x, point.y);
    }
}

class Swipe extends CustomEvent {
    private final Point start;
    private final Point end;

    public Swipe(Point start, Point end) {
        this.start = start;
        this.end = end;
    }

    @Override
    protected void createPath() {
        path.moveTo(start.x, start.y);
        path.lineTo(end.x, end.y);
    }
}

File: app/src/main/java/com/example/androidautoclicker/CustomEvent.kt
package com.example.androidautoclicker

import android.accessibilityservice.GestureDescription
import android.graphics.Path
import android.graphics.Point

sealed class CustomEvent(protected var startTime: Long = 10L, protected var duration: Long = 10L) {
    protected val path = Path()

    fun onEvent(): GestureDescription.StrokeDescription {
        path.reset()
        createPath()
        return GestureDescription.StrokeDescription(path, startTime, duration)
    }

    protected abstract fun createPath()
}

class Click(private val point: Point, startTime: Long = 10L, duration: Long = 10L) :
    CustomEvent(startTime, duration) {

    override fun createPath() {
        path.moveTo(point.x.toFloat(), point.y.toFloat()) // Use floats for moveTo
    }
}

class Swipe(private val start: Point, private val end: Point, startTime: Long = 10L, duration: Long = 10L) :
    CustomEvent(startTime, duration) {

    override fun createPath() {
        path.moveTo(start.x.toFloat(), start.y.toFloat())
        path.lineTo(end.x.toFloat(), end.y.toFloat())
    }
        }
